### PR TITLE
Send a Matrix notification when a BAD push is identified

### LINKED ISSUE #647: Send a Matrix message when a bad push is identified
Followup of #625 as only the email notification was implemented there.

@marco-c already created the `#sherrif-notifications` matrix room where  we should publish the same notifications n BAD pushes.

The configuration secret should have a new section as:

```
matrix-room-id = "#sherrif-notifications"
```

The local configuration `~/.config/mozci/config.toml` would then become

```
[mozci]
matrix-room-id = "#sherrif-notifications"
```

Here is the [API doc](https://docs.taskcluster.net/docs/reference/core/notify/api#matrix) to send notification on a given matrix room. It's using the same `notify` service as the email notification, with a different method.

Matrix does not [seem to support](https://spec.matrix.org/v1.1/client-server-api/#mroommessage-msgtypes) markdow directly, but would allow rendered HTML. We can make a first version that only specifies `body` using the existing markdown payload from the email (and no `formattedBody`)

### PR DESCRIPTION
**Pull request by EvaBardou**
Closes #647

### CODE REVIEW

File: mozci/console/commands/push.py [hunk 1]
```diff
@@ -226,26 +228,30 @@ def handle(self) -> None:
...
-                self.send_emails(emails, push, previous, classification, regressions)
+                self.send_notifications(
+                    emails, matrix_room, push, previous, classification, regressions
+                )
 
-    def send_emails(
+    def send_notifications(
         self,
-        emails: List[str],
+        emails: Union[List[str], None],
```
Comment (marco-c) [inline mozci/console/commands/push.py:246]:
```suggestion
        emails: Optional[List[str]],
```

File: mozci/console/commands/push.py [hunk 2]
```diff
@@ -226,26 +228,30 @@ def handle(self) -> None:
...
+        matrix_room: Union[str, None],
```
Comment (marco-c) [inline mozci/console/commands/push.py:247]:
```suggestion
        matrix_room: Optional[str],
```

File: mozci/console/commands/push.py [hunk 3]
```diff
@@ -226,26 +228,30 @@ def handle(self) -> None:
...
                     # We still want to send a notification if the current one is bad
                     previous = None
 
-                self.send_emails(emails, push, previous, classification, regressions)
+                self.send_notifications(
+                    emails, matrix_room, push, previous, classification, regressions
+                )
 
-    def send_emails(
+    def send_notifications(
```
Comment (marco-c) [inline mozci/console/commands/push.py:244]: Tests will need to be adjusted following this change

Comment (EvaBardou) [reply mozci/console/commands/push.py:244]: Oups I forgot that there were any

**Review by marco-c — APPROVED** [review]

**Pull request merged**